ipmanager/ip_manager.go

```diff
@@ -26,22 +28,27 @@ type IPManager struct {
 }
 
 // NewIPManager returns a new instance of IPManager
-func NewIPManager(hostingType string, config *IPConfiguration, states <-chan bool, verbose bool) (m *IPManager, err error) {
+func NewIPManager(config *vipconfig.Config, ipConfig *IPConfiguration, states <-chan bool) (m *IPManager, err error) {
```

Comment:
If it is not too complicated, we can get rid of the `IPConfiguration` struct altogether, I don't think there is much that the `vipconfig.Config` doesn't already contain or could be easily made to contain.

Reply:
So I've tried to remove `IPConfiguration` completely, but as it turns out this is way more work than really justifiable for this pull request. 
If we really want to clean that up, it could be done as a separate pull request.